src/test/java/org/example/Operations/CollectionOperationsTest.java

```diff
@@ -1,250 +1,483 @@
+        assertEquals("Removing collection request failed", 200, res.getHttpStatusCode());
+        assertEquals("Removing collection failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Stat the original collection to show that it does not exist.
+        res = rods.collections().stat(token, newCollectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat did not fail as expected", -170000,
+                getIrodsResponseStatusCode(res.getBody()));
+        // -170000: NOT_A_COLLECTION
+    }
+
+    @Test
+    public void testListOperation() {
+        String homeCollection = "/tempZone/home/alice";
+
+        // Created nested collections
+        String collection = homeCollection + "/c0/c1/c2";
+        res = alice.collections().create(aliceToken, collection, OptionalInt.of(1));
+        assertEquals("Creating collection request failed", 200, res.getHttpStatusCode());
+        assertEquals("Creating collection failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Create one data object in each collection.
+        List<String> dataObjects = new ArrayList<>();
+        dataObjects.add(homeCollection + "/c0/d0");
+        dataObjects.add(homeCollection + "/c0/c1/d1");
+        dataObjects.add(homeCollection + "/c0/c1/c2/d2");
+
+        for (String name : dataObjects) {
+            res = alice.dataObject().touch(aliceToken, name, null);
+            assertEquals("Creating data object request failed", 200, res.getHttpStatusCode());
+            assertEquals("Creating data object failed", 0,
+                    getIrodsResponseStatusCode(res.getBody()));
+        }
+
+        // List only the contents of the home collection.
+        res = alice.collections().list(aliceToken, homeCollection, null);
+        assertEquals("Listing collections request failed", 200, res.getHttpStatusCode());
+        String entry = null;
+        try {
+            JsonNode rootNode = mapper.readTree(res.getBody());
+            entry = rootNode.path("entries").get(0).asText();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        assertEquals("/tempZone/home/alice/c0", entry);
+
+        // List the home collection recursively.
+        CollectionsListParams listParams = new CollectionsListParams();
+        listParams.setRecurse(1);
+        res = alice.collections().list(aliceToken, homeCollection, listParams);
+        List<String> actualEntries = new ArrayList<>();
+        try {
+            JsonNode rootNode = mapper.readTree(res.getBody());
+            JsonNode entriesNode = rootNode.path("entries");
+
+            for (JsonNode node : entriesNode) {
+                actualEntries.add(node.asText());
+            }
+            Collections.sort(actualEntries);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
JUnit should offer an assertion for handling things that can throw. Prefer using that rather to try-catch.

Reply:
It's been replaced.